lib/checkother.cpp

```diff
@@ -1225,6 +1251,15 @@ bool CheckOther::checkInnerScope(const Token *tok, const Variable* var, bool& us
                         if (!ret.empty() && ret.back() == '*')
                             return false;
                     }
+                    if (ftok->function()) {
+                        const std::vector<const Token *> argtoks = getArguments(ftok);
```

Comment:
hmm I think you should use ftok->function()->argumentList() when you check the valuetypes.

The reference property is more interesting for the parameter declaration.